react-training/src/redux/auth/reducer.js

```diff
@@ -1,37 +1,20 @@
+import { createReducer, onLoading, onSuccess, onFailure } from 'redux-recompose';
+import Immutable from 'seamless-immutable';
+
 import { actionTypes } from '../auth/action';
 
-const initialState = {
-  isAuth: false,
-  user: null,
-  loggingIn: false
+const initialState = Immutable({ auth: null });
```

Comment:
Let's use [completeState](https://github.com/Wolox/redux-recompose/blob/master/src/completers/completeState/docs.md) for properties related to the api integration